src/lib/y2network/config_reader/sysconfig_interfaces.rb

```diff
@@ -39,51 +41,75 @@ module ConfigReader
     class SysconfigInterfaces
       # Returns the interfaces and connections configuration
       #
-      # @todo Should we use different readers?
-      # @todo Are virtual interfaces coming from connections only?
+      # @return [Hash<Symbol,Object>] Returns a hash containing
+      #   an interfaces collection (with the key `:interfaces`)
+      #   and an array of connection config objects.
       def config
-        [interfaces, connections]
+        return @config if @config
+        find_physical_interfaces
+        find_connections
+        { interfaces: @interfaces, connections: @connections }
```

Comment:
hmmm, I do not like much this magic with find and find. Can we simply load it when first called and have methods like interfaces and connections? Here it is localized at one place, so not big deal for me. BTW you also probably miss to assign `@config` here

Reply:
The trick is that we need to execute both methods in order to get a correct configuration. For sure, `find_physical_interfaces` should be called first, but that interfaces list could be augmented while reading the connection configurations (with fake interfaces).

So I just wanted to be sure that, if you call `#interfaces` (but not `config`), you get all the information read. Having a "#read" method (which is basically what this #config method does) solves the issue. Another approach might be to have separate methods for fake and physical interfaces (so getting fake interfaces implies reading connections but not the other way around), but I was simply not sure about this separation.